test/authentication.js

```diff
@@ -0,0 +1,289 @@
+const chai = require('chai');
+const expect = chai.expect;
+const sinon = require('sinon');
+
+//MODELS
+const { User } = require('../models');
+const bcrypt = require('bcrypt');
+const LoginController = require('../controllers/authentication');
+const RegisterController = require('../controllers/user-controller');
+const AuthDao = require('../dao/authentication');
+const Token = require('../helpers/Token');
+const NotifyViaEmail = require('../services/notifyViaEmail');
+
+describe('Authentication Unit Test', () => {
+  describe('Register Controller', () => {
+    context('Context of success', () => {
+      beforeEach(() => {
+        this.mockReq = {
+          body: {
+            firstName: 'test',
+            lastName: 'test',
+            email: 'test@test.com',
+            password: 'hash',
+          },
+        };
+
+        this.mockRes = {
+          json: function (body) {
+            this.body = body;
+            return this;
+          },
+          status: function (status) {
+            this.status = status;
+            return this;
+          },
+        };
+        this.stubToken = sinon.stub(Token, 'generateJWT').resolves('token');
+        this.stubBvrypt = sinon.stub(bcrypt, 'hashSync').resolves('hash');
+        this.stubDb = sinon.stub(User, 'create').resolves(this.mockReq.body);
+        this.stubEmail = sinon.stub(NotifyViaEmail, 'sendEmail').resolves(true);
+      });
+
+      afterEach(() => {
+        expect(this.mockRes.status).to.be.equal(200);
+        expect(this.mockRes.body).to.have.property('firstName');
+        expect(this.mockRes.body).to.have.property('lastName');
+        expect(this.mockRes.body).to.have.property('email');
+        expect(this.mockRes.body).to.have.property('token');
+        sinon.restore();
+      });
+
+      it('Should create a token using the generateJWT function with a email', async () => {
+        await RegisterController.post(this.mockReq, this.mockRes);
+        expect(this.stubToken.calledWith(this.mockReq.body.email)).to.be.true;
```

Comment:
nit: Todos los expects son del mismo test, no es necesario separar en varios IT ya que estás haciendo todo el flujo, no diferentes llamadas.